pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test: TestLogic_mixed_version_upgrade_preserve_ttl failed [logic test retry too short]

pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test.TestLogic_mixed_version_upgrade_preserve_ttl [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20515632?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20515632?buildTab=artifacts#/) on release-25.4 @ [9221c5318324b33455257088d3a48fae69346359](https://github.com/cockroachdb/cockroach/commits/9221c5318324b33455257088d3a48fae69346359):


```
=== RUN   TestLogic_mixed_version_upgrade_preserve_ttl
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/9f5fe453e1e6aefdb79fee60981c27dd/logTestLogic_mixed_version_upgrade_preserve_ttl2187296340
    test_log_scope.go:82: use -show-logs to present logs inline
    logic.go:3205: let $initial_version = 25.3
    logic.go:3143: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5763/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test_/cockroach-go-testserver-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:19: SELECT version != '25.3' FROM [SHOW CLUSTER SETTING version]
        expected:
            true
            
        but found (query options: "retry") :
            false
            
[22:08:15] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5763/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test_/cockroach-go-testserver-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl: 2 statements
[22:08:15] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5763/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test_/cockroach-go-testserver-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl with config cockroach-go-testserver-25.3: 2 tests, 1 failures
    logic.go:4494: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5763/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test_/cockroach-go-testserver-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:24: error while processing
    logic.go:4494: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5763/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test_/cockroach-go-testserver-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:24: too many errors encountered, skipping the rest of the input
cockroach logs captured in: /artifacts/tmp/_tmp/9f5fe453e1e6aefdb79fee60981c27dd/cockroach-logs1114556003
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9f5fe453e1e6aefdb79fee60981c27dd/logTestLogic_mixed_version_upgrade_preserve_ttl2187296340
--- FAIL: TestLogic_mixed_version_upgrade_preserve_ttl (93.54s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_mixed_version_upgrade_preserve_ttl.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54769

pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test.TestLogic_mixed_version_upgrade_preserve_ttl [failed](https://mesolite.cluster.engflow.com/invocations/default/8b6ede29-a2ea-4d22-9a90-08531a099311?testReportRun=6&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9jb2Nrcm9hY2gtZ28tdGVzdHNlcnZlci0yNS4zOmNvY2tyb2FjaC1nby10ZXN0c2VydmVyLTI1XzNfdGVzdA==) on release-25.4 @ [b3d4aa9739b2cc9961983d6dad862f5a6cdc08d9](https://github.com/cockroachdb/cockroach/commits/b3d4aa9739b2cc9961983d6dad862f5a6cdc08d9):


```
=== RUN   TestLogic_mixed_version_upgrade_preserve_ttl
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestLogic_mixed_version_upgrade_preserve_ttl1458111638
    test_log_scope.go:82: use -show-logs to present logs inline
    logic.go:3205: let $initial_version = 25.3
    logic.go:3143: 
         
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test_/cockroach-go-testserver-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:19: SELECT version != '25.3' FROM [SHOW CLUSTER SETTING version]
        expected:
            true
            
        but found (query options: "retry") :
            false
            
[05:53:29] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test_/cockroach-go-testserver-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl: 2 statements
[05:53:29] --- done: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test_/cockroach-go-testserver-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl with config cockroach-go-testserver-25.3: 2 tests, 1 failures
    logic.go:4494: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test_/cockroach-go-testserver-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:24: error while processing
    logic.go:4494: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test_/cockroach-go-testserver-25_3_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:24: too many errors encountered, skipping the rest of the input
cockroach logs captured in: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test/shard_8_of_8_run_6_of_10/test.outputs/cockroach-logs3327914132
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_mixed_version_upgrade_preserve_ttl1458111638
--- FAIL: TestLogic_mixed_version_upgrade_preserve_ttl (85.53s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #154550 pkg/sql/logictest/tests/cockroach-go-testserver-25.3/cockroach-go-testserver-25_3_test: TestLogic_mixed_version_upgrade_preserve_ttl failed [C-test-failure O-robot T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_mixed_version_upgrade_preserve_ttl.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl
+++ b/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl
@@ -16,7 +16,7 @@ upgrade all
 # Verify that the cluster version upgrades have begun by asserting we're no
 # longer on the previous version. Note that the first cluster upgrade is the
 # one that repairs all descriptors.
-query B retry
+query B retry retry_duration 90s
 SELECT version != '$initial_version' FROM [SHOW CLUSTER SETTING version]
 ----
 true

